AirCasting/MicrophoneSession/MicrophoneManager.swift

```diff
@@ -19,33 +20,37 @@ final class MicrophoneManager: NSObject, ObservableObject {
     private var measurementStreamLocalID: MeasurementStreamLocalID?
 
     //This variable is used to block recording more than one microphone session at a time
-    private(set) var isRecording = false
-    private var recorder: AVAudioRecorder!
+    private(set) var isRecording = false {
+        didSet { isRecording ? interruptionHandler.resume() : interruptionHandler.pause() }
+    }
+    private var recorder: AVAudioRecorder?
     private lazy var audioSession = AVAudioSession.sharedInstance()
     private var levelTimer: Timer?
     private(set) var session: Session?
     private lazy var locationProvider = LocationProvider()
+    private var interruptionHandler: AVSessionInterruptionHandler!
 
     init(measurementStreamStorage: MeasurementStreamStorage) {
         self.measurementStreamStorage = measurementStreamStorage
         super.init()
+        self.interruptionHandler = .init(observer: self, audioSession: audioSession)
+        do {
+            try recorder = AVAudioRecorder(url: URL(fileURLWithPath: "/dev/null", isDirectory: true), settings: MicrophoneManager.recordSettings)
+        } catch {
+            Log.error("Could not create AVAudioRecorder: \(error.localizedDescription)")
+        }
     }
 
     func startRecording(session: Session) throws {
-        if isRecording {
-            return
-        }
+        guard !isRecording, let recorder = recorder else { return }
         
         if audioSession.recordPermission != .granted {
             throw MicrophoneSessionError.permissionNotGranted
         }
         self.session = session
-
-        try audioSession.setCategory(AVAudioSession.Category.playAndRecord)
+        try setupAudioSession()
         try audioSession.setActive(true)
-        try recorder = AVAudioRecorder(url: URL(fileURLWithPath: "/dev/null", isDirectory: true), settings: MicrophoneManager.recordSettings)
         recorder.isMeteringEnabled = true
-        recorder.delegate = self
```

Comment:
This is not necessary?

Reply:
not anymore, delegate methods have been deprecated for some time and we use NotificationCenter notifs instead now